ksql-engine/src/main/java/io/confluent/ksql/codegen/SqlToJavaVisitor.java

```diff
@@ -539,11 +545,12 @@ private String visitBooleanComparisonExpression(final ComparisonExpression.Type
       switch (internalSchema.type()) {
         case ARRAY:
           return new Pair<>(
-              String.format("((%s) ((%s)%s).get((int)(%s)))",
+              String.format("((%s) (ArrayGet.getItem(((%s) %s), (int)(%s), %s)))",
```

Comment:
If index is null, then this will throw NPE.  The index may be derived from the source query e.g.

```
SELECT array_field[index_field] FROM ...
```

So we may want to consider handling a null index by just returning null. (I've not checked what other vendors do, but most seem to err on the side of not-logging-loads-of-errors and just returning null)

Reply:
If we throw an exception it will be logged in the processing log...